crates/solver/src/solver/http_solver.rs

```diff
@@ -445,6 +445,25 @@ fn compute_fee_connected_tokens(liquidity: &[Liquidity], native_token: H160) ->
     fee_connected_tokens
 }
 
+fn non_bufferable_tokens_used(
+    interactions: &[InteractionData],
+    market_makable_token_list: &HashSet<H160>,
+) -> Vec<H160> {
+    interactions
+        .iter()
+        .filter(|interaction| {
+            interaction
+                .exec_plan
+                .as_ref()
+                .map(|plan| plan.internalizable())
+                .unwrap_or_default()
+        })
+        .flat_map(|interaction| interaction.outputs.clone())
```

Comment:
@harisang using `outputs` vector here, where `outputs` is defined as:

> /// The output amounts from the AMM interaction - i.e. the amount of tokens
> /// that are expected to be sent from the AMM into the settlement contract
> /// for this calldata.

Just to make sure I understood the issue correctly.

Reply:
When we internalize, the settlement contract keeps in its buffers the buy token of the interaction, which we would otherwise have sent to the AMM. This means that an interaction can be internalized if its buy token is bufferable.

So, it seems to me you have to use the inputs vector and not the outputs vector.

Reply:
Correct, I think we need to use the `inputs` vector since it is the tokens that _would have_ gone to the AMM if the interaction wasn't internalized, but instead stays in the settlement contract: we keep it, so we need to trust it.

Also, ubernit, I don't think you need to clone here:
```suggestion
        .flat_map(|interaction| &interaction.inputs)
```

Reply:
Makes sense. Fixed.